src/Parser.php

```diff
@@ -216,20 +216,28 @@ function ($e) {
     }
 
     /**
-     * To avoid splitting on scoped package-names, every but the last @ are considered
-     * package name.
-     *
      * @param string $versionString
-     *
-     * @return string[]
+     * @return array{string, string}
+     * @throws ParserException
      */
     public static function splitVersionString($versionString)
     {
-         $parts = explode('@', $versionString);
-         $version = array_pop($parts);
-         return [
-             implode('@', $parts),
-             $version
-         ];
+        if (strpos($versionString, '@') === 0) {
+            $prefix = '@';
+            $rest = substr($versionString, 1);
+        } else {
+            $prefix = '';
+            $rest = $versionString;
+        }
+
+        $atPos = strpos($rest, '@');
+        if ($atPos === 0 || $atPos === false) {
+            throw new ParserException('Invalid version string: ' . $versionString);
```

Comment:
Could you please add a code to the exception as for the other ones, for easier identification, usually the current timestamp.

Reply:
Done, but only to restore consistency.
I recommend to remove those in the future though since I feel they offer no real benefit.
Exception types should be separated by exception name, grouped by interfaces, and provide details by message and possibly further properties (public, or exposed via getters).
Just my 2c